src/pugixml.cpp

```diff
@@ -6059,6 +6059,28 @@ namespace pugi
 		return true;
 	}
 
+	PUGI__FN bool xml_node::remove_attributes()
+	{
+		if (!_root) return false;
+		impl::xml_allocator& alloc = impl::get_allocator(_root);
+		if (!alloc.reserve()) return false;
+
+		for (xml_attribute_struct* attr = _root->first_attribute; attr;)
+		{
+			if (!attr) return false;			
```

Comment:
I believe these two checks are redundant (you will only ever see attributes of the node itself on the attribute list), but it’s also dangerous to return in the middle of this since some attributes were destroyed already.